### PR TITLE
content: add ai-foundations chunk for Wave 3 AI/ML probe

### PR DESCRIPTION
**Pull request by OpenSauce**
## Summary

Wave 3 probe content for the new **AI/ML Fundamentals** category. This PR ships a single lesson chunk + quiz + flashcards plus a STYLE.md author guide. It is a **2-week demand probe** — four additional chunks (neural-nets, transformers, embeddings, LLMs+RAG+evals) are gated on telemetry signals after this lands.

**Framing:** AI/ML interview vocabulary for non-ML engineers. Part of DSA Flash's "one app for all technical interview prep" story. Cards are tuned for "define this in 30 seconds on a whiteboard" retrieval, not long-form comprehension.

## What's in this PR

- `ai-ml-fundamentals/STYLE.md` — vocabulary pins (model, embedding, token, inference, etc.), tone rules, "never include" list (no GPT-4/Claude-3/Llama mentions, no context window sizes, no benchmark numbers, no vendor names), and schema foot-gun warnings for future content authors
- `ai-ml-fundamentals/lessons/ai-foundations.md` — one ~4-min lesson covering AI/ML/DL hierarchy, three flavors of learning, training vs inference, train/val/test split, loss & gradient descent intuition, overfitting
- `ai-ml-fundamentals/quizzes/ai-foundations.yaml` — 6 MCQs (each with exactly 4 options), linked to the lesson via `lesson_slug: ai-foundations`
- `ai-ml-fundamentals/ai-ml-fundamentals.yaml` — 8 flashcards, each linked to the lesson via `lesson: ai-foundations` (note the YAML key asymmetry with quizzes — documented in STYLE.md)

## Schema verification

- 8 flashcards, all with capitalized `Front`/`Back`, `lesson: ai-foundations`
- 6 quiz questions, every question has exactly 4 options, `lesson_slug: ai-foundations`
- Matches existing `system-design` category on-disk format

## Telemetry probe context

Parent repo PR (coming separately) ships the backend changes needed to measure this probe: per-category breakdowns in `/api/analytics/summary` and signup `referrer_category` attribution. Without that PR's analytics extension, the "ship and measure for 2 weeks" plan is not readable.

## Test plan

- [ ] Backend loader picks up all four files on container startup
- [ ] Category appears on homepage under "Other" section
- [ ] Lesson page renders, quiz submits
- [ ] Quiz completion seeds 8 UserFlashcard rows for the signed-in user
- [ ] No banned terms in cards, quiz options, or lesson body (manual spot-check)

### CODE REVIEW

File: ai-ml-fundamentals/lessons/ai-foundations.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+**Reinforcement learning** uses a reward signal instead of labels. An agent takes actions, receives a reward or penalty, and learns to act in ways that maximize long-term reward. Game-playing agents are the famous example. The model is not told what move is correct, only whether the outcome was good.
+
+## Training vs Inference
+
+This is the single most important distinction in ML system design. **Training** is the expensive process of fitting the model's parameters to data. It usually happens once, or periodically, and it can take hours, days, or weeks on specialized hardware. **Inference** is running the finished model on new input to get an answer. Inference is cheap per call, fast, and happens constantly in production. When someone asks "where does the cost live in your ML system", they are asking you to locate the training and inference workloads separately, because they have completely different scaling profiles, hardware needs, and latency requirements.
```
[the remark this answers is not in the document]
Comment (OpenSauce) [reply ai-ml-fundamentals/lessons/ai-foundations.md]: Applied in da21cd6. Lesson body now reads 'cheaper per call than training' with the 'major cost at scale' caveat, matching your suggestion.

File: ai-ml-fundamentals/quizzes/ai-foundations.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    explanation: "Unsupervised learning works on unlabeled data and finds structure on its own. Clustering users from raw usage logs is the canonical example. The spam filter and house-price predictor are supervised (labeled data), and the game-playing agent is reinforcement learning (reward signal)."
+
+  - question: "In an ML system, which workload is typically the expensive, one-time or periodic process, and which runs cheaply per call in production?"
+    options:
+      - "Training is cheap per call; inference is the expensive one-time process"
+      - "Training is the expensive one-time or periodic process; inference runs cheaply per call"
+      - "Both training and inference are equally expensive"
+      - "Training and inference are the same workload with different names"
+    correct: 1
+    explanation: "Training fits the model's parameters and can take hours to weeks on specialized hardware. Inference is running the finished model on new inputs and is cheap per call. This split is a fault line in ML system design because the two workloads have completely different scaling and hardware profiles."
```
[the remark this answers is not in the document]
Comment (OpenSauce) [reply ai-ml-fundamentals/quizzes/ai-foundations.yaml]: Applied in da21cd6. Quiz Q2 question text, options, and explanation all updated to 'cheaper per call than training' framing.

File: ai-ml-fundamentals/ai-ml-fundamentals.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- title: "AI Foundations - Training vs Inference"
+  difficulty: "medium"
+  tags: ["ml-basics", "training", "inference"]
+  lesson: ai-foundations
+  Front: |
+    In ML systems, what is the difference between training and inference, and why does it matter for system design?
+  Back: |
+    **Training:** fitting the model's parameters to data. Expensive, one-time or periodic, often hours to weeks on specialized hardware.
+
+    **Inference:** running the trained model on new input to get an answer. Cheap per call, fast, happens constantly in production.
```
[the remark this answers is not in the document]
Comment (OpenSauce) [reply ai-ml-fundamentals/ai-ml-fundamentals.yaml]: Applied in da21cd6. Flashcard Back text updated to 'typically cheaper per call than training ... can still be a major cost at scale' to keep the interview answer accurate.

File: ai-ml-fundamentals/lessons/ai-foundations.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+---
+title: "AI/ML Foundations"
+summary: "The vocabulary every non-ML engineer needs: supervised vs unsupervised, training vs inference, overfitting, and why model choice is a tradeoff."
```
[the remark this answers is not in the document]
Comment (OpenSauce) [reply ai-ml-fundamentals/lessons/ai-foundations.md]: Applied in da21cd6. Removed 'and why model choice is a tradeoff' from the summary since the body does not cover model selection tradeoffs. Keeping the lesson tight is preferable to adding scope mid-probe.
